osc/util/debquery.py

```diff
@@ -51,29 +51,15 @@ def read(self, all_tags=False, self_provides=True, *extra_tags):
             raise DebError(self.__path, 'no debian binary')
         if debbin.read() != b'2.0\n':
             raise DebError(self.__path, 'invalid debian binary format')
-        control = arfile.get_file(b'control.tar.gz')
-        if control is not None:
-            # XXX: python2.4 relies on a name
-            tar = tarfile.open(name='control.tar.gz', fileobj=control)
-        else:
-            control = arfile.get_file(b'control.tar.xz')
-            if control:
-               if not HAVE_LZMA:
-                   raise DebError(self.__path, 'can\'t open control.tar.xz without python-lzma')
-               decompressed = lzma.decompress(control.read())
-               tar = tarfile.open(name="control.tar.xz",
-                                  fileobj=BytesIO(decompressed))
-            else:
-                control = arfile.get_file(b'control.tar.zst')
-                if control:
-                    if not HAVE_ZSTD:
-                        raise DebError(self.__path, 'can\'t open control.tar.zst without python-zstandard')
-                    with zstandard.ZstdDecompressor().stream_reader(BytesIO(control.read())) as reader:
-                        decompressed = reader.read()
-                    tar = tarfile.open(name="control.tar.zst",
-                                       fileobj=BytesIO(decompressed))
-            if control is None:
-                raise DebError(self.__path, 'missing control.tar')
+        for open_func in [self.__open_tar_gz,
+                          self.__open_tar_xz,
+                          self.__open_tar_zst,
+                          self.__open_tar]:
+            tar = open_func(arfile)
+            if tar is not None:
+                break
+        if tar is None:
+            raise DebError(self.__path, 'missing control.tar')
         try:
             name = './control'
             # workaround for python2.4's tarfile module
```

Comment:
This is probably not needed as we support python 3.6+

Reply:
I removed all of the python 2.x stuff for compatibility in395df87f.